services/worker-manager/src/worker-scanner.js

```diff
@@ -63,6 +64,16 @@ class WorkerScanner {
           this.monitor.info(
             `Worker ${worker.workerGroup}/${worker.workerId} has unknown providerId ${worker.providerId} (ignoring)`);
         }
+
+        // If the worker will be expired soon but it still exists,
+        // update it to stick around a while longer. If this doesn't happen,
+        // long-lived instances become orphaned from the provider. We don't update
+        // this on every loop just to avoid the extra work when not needed
+        if (worker.expires < taskcluster.fromNow('1 week')) {
+          await worker.update(this.db, worker => {
+            worker.expires = taskcluster.fromNow('1 week');
```

Comment:
(Somehow I commented on the commit and not the PR)

Change this to `8 days` and I think this is good to go.  The reason is that we only want to occasionally bump the expires value, rather than on every scan -- so we bump it high enough that the next scan (in fact, the next day's worth of scans) won't satisfy the `< 1 week` condition.